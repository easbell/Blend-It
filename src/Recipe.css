.recipe-stuff {
  display: inline-flex;
}

img {
  width: 10rem;
  height: 10rem;
} 