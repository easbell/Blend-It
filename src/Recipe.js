import React, { Component } from 'react';

export default function Recipe(props) {
  return (
    <div className='recipe-details'>
        <h4>Servings: {props.chosenRecipe.servings}</h4>
        <h4><a href={props.chosenRecipe.source}>Source</a></h4>
      {
        props.chosenRecipe.recipe.map(steps => {
          return (
              <li key={steps}>{steps}</li>
          )
        })
      }
    </div>
  )
}

