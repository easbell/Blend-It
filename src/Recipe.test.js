import React from 'react';
import Recipe from './Recipe.js';
import { shallow } from 'enzyme'

const mockRecipe = [
  "1 cup strawberries",
  "1/2 cup pineapple",
  "1 banana",
  "2 cups orange juice",
  "1/2 cup greek yogurt",
  "1 cup spinach, optional",
  "1 tablespoon chia or flaxseed, optional",
  "ice"
]

describe('Recipe', () => {
  const wrapper = shallow(
    <Recipe recipe={mockRecipe}/>
  );

  it('should match a snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })
})