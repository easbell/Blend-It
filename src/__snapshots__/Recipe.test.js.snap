// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe should match a snapshot with all data passed in 1`] = `
<div
  className="recipe-details"
>
  <h4>
    Servings: 
    4
  </h4>
  <h4>
    <a
      href="https://www.modernhoney.com/6-healthy-superfood-smoothies/"
    >
      Source
    </a>
  </h4>
  <ul>
    <li
      key="1 cup strawberries"
    >
      1 cup strawberries
    </li>
    <li
      key="1/2 cup pineapple"
    >
      1/2 cup pineapple
    </li>
    <li
      key="1 banana"
    >
      1 banana
    </li>
    <li
      key="2 cups orange juice"
    >
      2 cups orange juice
    </li>
    <li
      key="1/2 cup greek yogurt"
    >
      1/2 cup greek yogurt
    </li>
    <li
      key="1 cup spinach, optional"
    >
      1 cup spinach, optional
    </li>
    <li
      key="1 tablespoon chia or flaxseed, optional"
    >
      1 tablespoon chia or flaxseed, optional
    </li>
    <li
      key="ice"
    >
      ice
    </li>
  </ul>
</div>
`;